In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from datetime import datetime, timedelta
import warnings
warnings.filterwarnings('ignore')

# Import the refactored TrendAnalyzer
from soros_system.main import TrendAnalyzer

# Define asset IDs and paths to your data
asset_ids = [
    'bitcoin', 'ethereum', 'binancecoin',
    # Add more assets as needed
]

data_path='/Users/valter.rebelo/MissionControl/data/micro/candleData/'
btc_data_path='/Users/valter.rebelo/MissionControl/data/micro/candleData/bitcoin_candles.csv'
ssr_data_path='/Users/valter.rebelo/MissionControl/data/onchainData/BTC_SSR.csv'

# Load Markov volatility model if you have one
markov_model_name = '20250325_144631' # Optional

# Initialize the TrendAnalyzer
analyzer = TrendAnalyzer(
    asset_ids=asset_ids,
    data_path=data_path,
    btc_data_path=btc_data_path,
)

# Analyze all assets
results = analyzer.analyze_multiple_assets()

latest_trends = analyzer.get_latest_trends()
print(latest_trends)

2025-03-27 18:03:28,829 - INFO - soros_system.main - Initializing TrendAnalyzer...
2025-03-27 18:03:29,275 - INFO - soros_system.main - Initialized DataLoader with 3 assets
2025-03-27 18:03:29,278 - INFO - soros_system.main - TrendAnalyzer initialization complete
2025-03-27 18:03:29,278 - INFO - soros_system.main - Analyzing 3 assets...
2025-03-27 18:03:29,278 - INFO - soros_system.main - Analyzing asset: bitcoin
2025-03-27 18:03:29,284 - INFO - soros_system.data.data_loader - Successfully loaded data for bitcoin with 4350 rows.
2025-03-27 18:03:34,392 - WARNING - soros_system.main - No 'bitcoin_btc' column found for bitcoin, BTC trend analysis will be skipped
2025-03-27 18:03:34,393 - INFO - soros_system.indicators.trend_classifier - Processing USD trend data for bitcoin
2025-03-27 18:03:36,547 - INFO - soros_system.indicators.trend_classifier - bitcoin USD trend coverage: 99.17%
2025-03-27 18:03:36,550 - INFO - soros_system.main - bitcoin trend classification - USD: True, BTC: False


  ticker       date     close  Trend_Short Term_USD  Trend_Medium Term_USD  \
0    BTC 2025-03-25  87521.00                   1.0                   -1.0   
1    ETH 2025-03-25   2068.60                   1.0                   -1.0   
2    BNB 2025-03-25    630.31                   1.0                    0.0   

   Trend_Long Term_USD  Overall_Trend_USD  Trend_Short Term_BTC  \
0                  0.0                0.0                   NaN   
1                 -1.0               -1.0                   1.0   
2                  0.0                0.0                  -1.0   

   Trend_Medium Term_BTC  Trend_Long Term_BTC  Overall_Trend_BTC  
0                    NaN                  NaN                NaN  
1                   -1.0                 -1.0               -1.0  
2                    1.0                 -1.0                0.0  


In [4]:
results.get('ethereum')

,date,open,high,low,close,ethereum_btc_open,ethereum_btc_high,ethereum_btc_low,ethereum_btc,asset_id,...,MA252_ethereum_btc,RoC252_ethereum_btc,MA365_ethereum_btc,RoC365_ethereum_btc,Trend_Short Term_BTC,Trend_Medium Term_BTC,Trend_Long Term_BTC,Overall_Trend_BTC,returns_usd,returns_btc
0,2015-08-07,2.830000,2.830000,2.830000,2.830000,0.010161,0.010161,0.010161,0.010161,ethereum,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,2015-08-08,1.330000,1.330000,1.330000,1.330000,0.005119,0.005119,0.005119,0.005119,ethereum,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,-0.530035,-0.496190
2,2015-08-09,1.008793,1.008793,1.008793,1.008793,0.003816,0.003816,0.003816,0.003816,ethereum,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,-0.241509,-0.254536
3,2015-08-10,0.687586,0.687586,0.687586,0.687586,0.002609,0.002609,0.002609,0.002609,ethereum,...,NaN,NaN,NaN,NaN,-1.0,NaN,NaN,NaN,-0.318407,-0.316442
4,2015-08-11,1.067000,1.067000,1.067000,1.067000,0.003954,0.003954,0.003954,0.003954,ethereum,...,NaN,NaN,NaN,NaN,-1.0,NaN,NaN,NaN,0.551806,0.515637
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3514,2025-03-21,1985.520000,1993.580000,1938.900000,1964.530000,0.023546,0.023518,0.023293,0.023384,ethereum,...,0.036887,-56.784343,0.041425,-56.131219,0.0,-1.0,-1.0,-1.0,-0.008229,-0.005148
3515,2025-03-22,1965.250000,2002.850000,1965.250000,1979.540000,0.023383,0.023705,0.023477,0.023624,ethereum,...,0.036768,-55.965323,0.041346,-54.848119,0.0,-1.0,-1.0,-1.0,0.007641,0.010250
3516,2025-03-23,1980.460000,2019.120000,1980.410000,2001.050000,0.023627,0.023526,0.023626,0.023326,ethereum,...,0.036649,-56.297049,0.041267,-55.283652,-1.0,-1.0,-1.0,-1.0,0.010866,-0.012642
3517,2025-03-24,2007.210000,2102.940000,1981.370000,2077.740000,0.023309,0.023705,0.023158,0.023792,ethereum,...,0.036530,-55.781552,0.041192,-53.637301,1.0,-1.0,-1.0,-1.0,0.038325,0.020014


In [5]:
latest_trends

,ticker,date,close,Trend_Short Term_USD,Trend_Medium Term_USD,Trend_Long Term_USD,Overall_Trend_USD,Trend_Short Term_BTC,Trend_Medium Term_BTC,Trend_Long Term_BTC,Overall_Trend_BTC
0,BTC,2025-03-25,87521.00,1.0,-1.0,0.0,0.0,NaN,NaN,NaN,NaN
1,ETH,2025-03-25,2068.60,1.0,-1.0,-1.0,-1.0,1.0,-1.0,-1.0,-1.0
2,BNB,2025-03-25,630.31,1.0,0.0,0.0,0.0,-1.0,1.0,-1.0,0.0


the result of analyzer.get_latest_trends() also doesn't make much sense: 

asset_id	date	close	Trend_Short Term_USD	Trend_Medium Term_USD	Trend_Long Term_USD	Overall_Trend_USD	Overall_Trend_BTC	ethereum_btc	Trend_Short Term_BTC	Trend_Medium Term_BTC	Trend_Long Term_BTC	binancecoin_btc	ripple_btc	cardano_btc
0	bitcoin	2025-03-25	87521.000000	2.0	-2.0	0.0	0.0	NaN	NaN	NaN	NaN	NaN	NaN	NaN	NaN
1	ethereum	2025-03-25	2068.600000	NaN	NaN	NaN	NaN	-1.0	0.023635	2.0	-2.0	-2.0	NaN	NaN	NaN

Important: DO NOT USE asset ids, these should be used only at input if necessary. Get the tickers instead. there's a method for this. 
